src/QsCompiler/CommandLineTool/Commands/Build.cs

```diff
@@ -74,7 +96,8 @@ internal static bool IncorporateResponseFiles(BuildOptions options, out BuildOpt
                     try
                     {
                         var fromResponseFiles = FromResponseFiles(options.ResponseFiles);
-                        if (fromResponseFiles == null) return false;
+                        if (fromResponseFiles == null)
```

Comment:
Is this because an instruction can't be in-lined with an `if/else if/else`?

If that is the case, I think it would be better to always enclose instructions after an `if/else if/else` in curly brakets `{ }` because otherwise it is more likely that an error is introduced.

Reply:
Hmm, I think at some point having the statement on the same line as an `if` wasn't allowed, but now it seems like it is... Maybe I changed some StyleCop settings since then. However, I can't disallow the two-line version without also disallowing the one-line version. (There is an open issue for that: DotNetAnalyzers/StyleCopAnalyzers#2252.)

I think that omitting braces is unlikely to cause errors, because

```
if (foo)
    Bar();
    Baz();
```

will produce a warning from [SA1137: Elements Should Have the Same Indentation](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1137.md).

Reply:
Although the two-line version is allowed with the current style rules, I reverted all of them back to one-line versions to avoid making unnecessary formatting changes in this PR. Are we OK with this ambiguity in the style rules (either one-line or two-line braceless `if`s are allowed)?

Reply:
My preference is to allow single-line and braced multi-line, but disallow unbraced two-line. However, I am fine with allowing the unbraced two-line if statements if people really want them.

Reply:
The problem is that StyleCop's rule settings are not precise enough, so we can't allow unbraced one-line without also allowing unbraced two-line, even if no one wants them.

Reply:
Oh I see the problem. In that case I can see the desire to not want to have a rule for this and allow all the options.

Reply:
Given that the SA1137 warning will be produced, I think it is ok to have this ambiguity in the style rules.

Reply:
I figured out why this was not allowed earlier: the change from a ruleset file to an editorconfig file meant that StyleCop no longer realized that SA1503 (braces) was disabled, which meant that SA1501 (statement must be on single line) was behaving differently - it doesn't produce its own warning in this case because it thinks the braces warning will appear, but that was suppressed by editorconfig.

I would actually like to go back to that behavior (by re-adding a ruleset just to disable SA1503) and require that ifs be written on at least two lines in all cases. While I'm fine with short lines like

```
if (foo is null) throw new ArgumentNullException(nameof(foo));
```

I think it's easy for this to get out of hand. For example, these lines are from DiagnosticTools:

```
if (relativeRange == null) throw new ArgumentNullException(nameof(relativeRange));
if (LargerThan(relativeRange.Item1, relativeRange.Item2)) throw new ArgumentException("invalid range", nameof(relativeRange));
```

This code is very dense and the `throw`s do not line up, so it's hard to scan the code and understand the structure of it, IMO. I think that writing it like this is easier to read:

```
if (relativeRange == null)
    throw new ArgumentNullException(nameof(relativeRange));
if (LargerThan(relativeRange.Item1, relativeRange.Item2))
    throw new ArgumentException("invalid range", nameof(relativeRange));
```

Braces would be OK too, but when there are a lot of throws at the top of a method, having braces makes everything twice as long which also has its readability problems, I think. Is everyone OK with this?

Reply:
Preventing single line if statements seems fine to me. This style rule seems fairly common too, base on my experiences.

Reply:
I slightly prefer having braces all the time but I would be ok with not requiring them all the time of most people prefer it that way.

Reply:
either one is fine by me

Reply:
It seems like requiring braces is most people's first choice. It's also the default setting for StyleCop. So requiring braces seems like the option with the least friction overall. I'll change the settings to use that way.